tests/cuda/test_thom.f90

```diff
@@ -91,64 +89,85 @@ program test_thom
   u_dev = u
 
   ! preprocess the operator and coefficient arrays
-  tdsops = cuda_tdsops_init(n, dx, operation='second-deriv', &
-                            scheme='compact6', &
+  tdsops = cuda_tdsops_init(n, dx, & 
+                            operation='second-deriv', scheme='compact6', &
                             bc_start=BC_DIRICHLET, bc_end=BC_DIRICHLET)
+
   call cpu_time(tstart)
   do i = 1, n_iters
     call exec_thom_tds_compact(du_dev, u_dev, tdsops, blocks, threads)
   end do
   call cpu_time(tend)
   print *, 'Total time', tend - tstart
 
-  ! 2 in fw pass, 2 in bw pass: 4 in total
-  call checkperf(tend - tstart, n_iters, ndof, 4._dp)
-
-  ! check error
+  ! move data to host
   du = du_dev
-  norm_du = norm2(u + du)
-  norm_du = norm_du*norm_du/n_glob/n_block/SZ
-  norm_du = sqrt(norm_du)
 
-  print *, 'error norm', norm_du
+  ! 2 in fw pass, 2 in bw pass: 4 in total
+  call checkperf(tend - tstart, n_iters, ndof, 4._dp)
+  call checkerr(u, du, 1.0e-8_dp)
 
-  if (norm_du > tol) then
-    allpass = .false.
-    write (stderr, '(a)') 'Check dirichlet second derivatives... failed'
-  else
-    write (stderr, '(a)') 'Check dirichlet second derivatives... passed'
-  end if
   if (allpass) then
-    write (stderr, '(a)') 'ALL TESTS PASSED SUCCESSFULLY.'
+      print *, 'ALL TESTS PASSED SUCCESSFULLY.'
   else
     error stop 'SOME TESTS FAILED.'
   end if
 
 contains
 
-  subroutine checkperf(t_tot, n_iters, ndof, consumed_bw)
+  subroutine checkperf(trun, n_iters, ndof, consumed_bw)
     implicit none
 
-    real(dp), intent(in) :: t_tot, consumed_bw
-    integer, intent(in) :: n_iters, ndof
+    real(kind(0.0d0)), intent(in) :: trun
+    integer, intent(in) :: n_iters
+    integer, intent(in) :: ndof
+    real(dp), intent(in) :: consumed_bw
 
-    real(dp) :: achievedBW, devBW
-    integer :: ierr, memClockRt, memBusWidth
-
-    ! BW utilisation and performance checks
-    achievedBW = consumed_bw*n_iters*ndof*dp/t_tot
-
-    print'(a, f8.3, a)', 'Achieved BW: ', achievedBW/2**30, ' GiB/s'
+    integer :: ierr
+    integer :: nbytes, memClockRt, memBusWidth
+    real(dp) :: achievedBW, deviceBW
 
+    if (dp == kind(0.0d0)) then
+      nbytes = 8
+    else
+      nbytes = 4
+    end if
```

Comment:
Is it worth making this a constant, e.g. where `dp` is defined? I can imagine there might be other uses of `nbytes` elsewhere